src/spi.rs

```diff
@@ -432,30 +497,17 @@ impl<'d> SpiMasterDriver<'d> {
                 SPI_DEVICE_NO_DUMMY
             } else {
                 0_u32
+            } | if config.inverted_chip_select {
+                SPI_DEVICE_POSITIVE_CS
+            } else {
+                0_u32
             } | config.duplex.as_flags(),
             ..Default::default()
-        };
-
-        let mut device_handle: spi_device_handle_t = ptr::null_mut();
-
-        esp!(unsafe {
-            spi_bus_add_device(SPI::device(), &device_config, &mut device_handle as *mut _)
-        })?;
-
-        Ok(Self {
-            host: SPI::device() as _,
-            device: device_handle,
-            max_transfer_size: config.dma.max_transfer_size(),
-            _p: PhantomData,
-        })
-    }
-
-    pub fn device(&self) -> spi_device_handle_t {
-        self.device
+        }
     }
 
-    pub fn host(&self) -> spi_host_device_t {
-        self.host as _
+    pub fn cs_gpio_number(&self) -> i32 {
```

Comment:
`host` and `device` methods seem to be removed now? Or did I misread? From my POV they were there so that in case you want to use the Rust SPI driver for the most part, but then just need "this little thing" to be done with an unsafe call to the actual `esp-idf-sys` raw bindings - you could still do this. I see less of a need to do this for the CS pin, as the CS pin is passed *from user code*, so I would retire `cs_gpio_number` and restore the other two methods.

Reply:
included again